### DISCUSSION TITLE
360 minutes as default timout for github actions is too large

### DISCUSSION THREAD

**Discussion by timotheecour** [community/community#4109] [Actions]
this caused all of our CI to be stuck in queuing because 1 bad commit was causing a test to hang forever; the fix was to revert that commit and then manually go through all queued/in-process jobs to cancel / restart them, and then change CI with `timeout-minutes: 60` to avoid this in the future, but IMO the default of 360 minutes is too large; this problem would've been discovered faster with a saner default of 60 minutes, which for most CI code is sufficient.

refs https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes

## note
for azure pipelines, the default timeout is 60mn, refs https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=azure-devops&tabs=yaml, a much saner default

**Comment by Levi-Lesches** [comment]
If this is implemented, there should be an option (or default) to keep the 360 limit. I predict that lowering a timeout on a platform with so many users can cause many issues.

**Reply by chrispat** [reply]
When we built Azure Pipelines, we set the default of 60 minutes which worked well for the private usage scenarios. As we opened it up for public repositories and advertised the maximum runtime of 360 minutes per job we saw many support requests from users whose builds were timing out at 60 until the explicitly changed the timeout on the job.  

When we built GitHub Actions, we decided to have the default be the maximum to prevent that confusion and acknowledged the tradeoff we were making.  

I think there is room for an option to let the user set the default timeout for a given org or repository if that makes sense for them.

**Reply by Levi-Lesches** [reply]
Exactly. Maybe something like a job-wide `timeout` property in each `job` (and a corresponding property under [`defaults`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaults)) if people want to explicitly lower it.
